app/src/lib/stores/app-store.ts

```diff
@@ -825,19 +825,40 @@ export class AppStore extends TypedBaseStore<IAppState> {
       gitStore.tip.kind === TipState.Valid &&
       repository.gitHubRepository !== null
     ) {
+      const gitHubRepo = repository.gitHubRepository
       const branchName = findRemoteBranchName(
         gitStore.tip,
         gitStore.currentRemote,
-        repository.gitHubRepository
+        gitHubRepo
       )
 
       if (branchName !== null) {
-        const currentBranchProtected = await this.repositoriesStore.isBranchProtectedOnRemote(
-          repository.gitHubRepository,
-          branchName
+        const account = getAccountForEndpoint(
+          this.accounts,
+          gitHubRepo.endpoint
         )
+
+        if (account === null) {
+          return
+        }
+
+        const name = gitHubRepo.name
+        const owner = gitHubRepo.owner.login
+        const api = API.fromAccount(account)
+
+        const {
+          allow_actor,
+          required_status_checks,
+          required_approving_review_count,
+        } = await api.fetchPushControl(owner, name, branchName)
+
+        const pushableByUser =
+          allow_actor &&
+          required_status_checks.length === 0 &&
+          required_approving_review_count === 0
```

Comment:
could this go into a helper method that lives somewhere in `lib/`? this is pretty crucial logic and i want to make sure its clearly encapsulated and documented.

i'm also a little unclear why we need to use the required status checks and review counts. are these enforced for pushing directly to a branch and not just for merges?